src/js/media-time-range.js

```diff
@@ -148,6 +152,17 @@ class MediaTimeRange extends MediaChromeRange {
   //   // TODO: Reset value after media is unset
   // }
 
+  get mediaDuration() {
+    return +this.getAttribute(MediaUIAttributes.MEDIA_DURATION);
+  }
+
+  get mediaBuffered() {
+    const buffered = this.getAttribute(MediaUIAttributes.MEDIA_BUFFERED);
+    if (buffered) {
+      return buffered.split(' ').map((timePair) => timePair.split(','));
+    }
+  }
+
```

Comment:
blocking: since this will suddenly add buffered indication to the time bar, I'm a bit concerned about rolling it out with the current color logic (found just below):

```js
const buffPercent =
      (buffered[buffered.length - 1][1] / this.mediaDuration) * 100;
    colorsArray.splice(1, 0, [
      'var(--media-time-buffered-color, #777)',
      buffPercent,
    ]);
```

This uses the very last `end` value and will render as if everything before it is buffered. Do we want that to suddenly be the default visualization for everyone?

Reply:
I didn't touch this logic but yes might need some changes.

I would suggest make it the end of the buffer range where the playhead currently is in, should be less confusing if they would hit a buffer stall...